hack/windows/Install-OVS.ps1

```diff
@@ -187,31 +226,219 @@ function InstallOVS() {
         exit 1
     }
-    $OVS_BIN_PATH="$OVSInstallDir\usr\bin;$OVSInstallDir\usr\sbin"
-    $env:Path += ";$OVS_BIN_PATH"
-    SetEnvVar "Path" $env:Path
+    Log "OVSExt driver has been installed"
+}
+
+function getInstalledVcRedists {
+    # Get all installed Visual C++ Redistributables installed components
+    $VcRedists = listInstalledSoftware -SoftwareLike 'Microsoft Visual C++'
+
+    # Add Architecture property to each entry
+    $VcRedists | ForEach-Object { If ( $_.Name.ToLower().Contains("x64") ) `
+        { $_ | Add-Member -NotePropertyName "Architecture" -NotePropertyValue "x64" } }
+
+    return $vcRedists
 }
 
-function InstallDependency() {
+function listInstalledSoftware {
+    param (
+        [parameter(Mandatory = $false)] [string] $SoftwareLike
+    )
+    Begin {
+        $SoftwareOutput = @()
+        $InstalledSoftware = (Get-ItemProperty -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\*)
+    }
+    Process {
+        Try
+        {
+            if ($SoftwareLike -ne "") {
+                $nameFilter = "${SoftwareLike}*"
+                $InstalledSoftware = $InstalledSoftware |
+                        Where-Object {$_.DisplayName -like "$nameFilter"}
+            }
+
+            $SoftwareOutput = $InstalledSoftware |
+                    Select-Object -Property @{
+                        Name = 'Date Installed'
+                        Exp  = {
+                            $_.Installdate
+                        }
+                    }, @{
+                        Name = 'Version'
+                        Exp  = {
+                            $_.DisplayVersion
+                        }
+                    }, @{
+                        Name = 'Name'
+                        Exp = {
+                            $_.DisplayName
+                        }
+                    }, UninstallString
+        }
+        Catch
+        {
+            # get error record
+            [Management.Automation.ErrorRecord]$e = $_
+
+            # retrieve information about runtime error
+            $info = New-Object -TypeName PSObject -Property @{
+                Exception = $e.Exception.Message
+                Reason    = $e.CategoryInfo.Reason
+                Target    = $e.CategoryInfo.TargetName
+                Script    = $e.InvocationInfo.ScriptName
+                Line      = $e.InvocationInfo.ScriptLineNumber
+                Column    = $e.InvocationInfo.OffsetInLine
+            }
+
+            # output information. Post-process collected info, and log info (optional)
+            $info
+        }
+    }
+
+    End{
+        $SoftwareOutput | Sort-Object -Property Name
+    }
+}
+
+function getInstalledOVSDrivers {
+    $pnputilOutput = pnputil.exe /enum-drivers
+    $drivers = @()
+    $lines = $pnputilOutput -split "`r`n"
+    $driverlines = @()
+    foreach ($line in $lines) {
+        if ($line -like "*Microsoft PnP Utility*") {
```

Comment:
could you add a comment explaining this check?

Reply:
Sure, I will add comments.  

The fact is this is  the title line of the output of command "pnputil.exe /enum-drivers".  The output of the command is not object-structured, which introduces troubles to use driver name as filters. So in the script, I am trying to translate them to objects firstly, then use the ovsext driver name as filters to check the existence. The title line is useless to translate the objects, so I ignore it here.